linuxNew/README.md

```diff
@@ -1,10 +1,80 @@
-# Linux Packages of Eclipse Adoptium 
+# Linux Packages of Eclipse Adoptium
 
-## RPMs for Fedora/Red Hat
+We package for Debian, Red Hat, Suse (e.g. Yum and RPM based) Linux distributions.
 
-### Other Topics
+The current implementation to build the packages involves using Gradle to call a small Java program.
+That Java program spinning up a Docker container, installing the base O/S and its packaging tools,
+and then looping over configuration to create the various packages and sign them as appropriate.
 
-### Building SRPMs and RPMs Directly
+TODO You can then optionally upload those packages to a package repository of your choice.  As a default
+the scripts upload to the Eclipse Foundation Nexus Package Repository as part of the Eclipse Adoptium
+CI/CD pipeline runs.
+
+## Prerequisites
+
+To run this locally
+
+* You will need to have Docker installed and running.
+* You will need to have Java 8+ installed.
+* You will need to have a minimum of 16GB of RAM on your system.
```

Comment:
Is 16GB a hard requirement? I've run containers on systems with far less and I don't think we're doing anything too intensive here.

Reply:
The problem is TestContainers. While copying files into containers, it creates multiple copies of the same file in memory. This is not a problem if you copy 2 MB files, but it becomes one with the JDKs that are nowadays more than 200 MB large. I had to set `-Xmx=4G` during development. 16 GB seems a bit excessive, but might happen in Gradle's `--parallel` mode.

I dug into the TestContainers code to find the problem. Somebody would have to patch it to use piped streams, but that requires a separate thread for copying the bytes. Because TestContainer's support of other platforms than x64 is somewhat weak (It's like using plain Docker) and we don't use any of its advanced capabilities, I'd replace TestContainer with plain Docker. Apart from lowering the memory consumption it suddenly becomes possible to use podman which would future-proof the tests.